jhub_apps/config_utils.py

```diff
@@ -48,3 +48,8 @@ class JAppsConfig(SingletonConfigurable):
         2,
         help="The number of workers to create for the JHub Apps FastAPI service",
     ).tag(config=True)
+
+    allow_multiple_jupyterlab = Bool(
+        False,
+        help="Allow users to spinup multiple JupyterLab servers",
+    ).tag(config=True)
```

Comment:
I wonder if we could improve naming here:
- I have not tried it but knowing the codebase, it feels like user could spin up multiple JupyterLab servers when this is `False` by using JupyterHub API directly (if named servers are not disabled)
- I wonder if we should instead name the option based on security implications; unfortunately there are multiple implications here so a single boolean would not suffice; I am thinking about:
  - a) "allow users to swap jupyterlab runtime" (potentially disabling system extensions)
  - b) "allow users to share their entire filesystem"

We could have separate options for (a) and (b) though but these would require different implementation. If that is not desirable, we could name this option "allow_jupyterlab_framework" as this is what it does. Maybe even more broadly it should be `allowed_frameworks`/`blocked_frameworks` list?

Reply:
>I wonder if we could improve naming here:

I agree with you. I was struggling to find a more apt name for this.

>I have not tried it but knowing the codebase, it feels like user could spin up multiple JupyterLab servers when this is False by using JupyterHub API directly (if named servers are not disabled)

Yep, correct. We only disabled it from jhub-apps interface and jhub-apps service API.

>I wonder if we should instead name the option based on security implications; unfortunately there are multiple implications here so a single boolean would not suffice; I am thinking about:
a) "allow users to swap jupyterlab runtime" (potentially disabling system extensions)
b) "allow users to share their entire filesystem"

Since it's an implication rather than the feature, I am not very confident on naming it like that. Although documentation would be an excellent place for this, wdyt?

>If that is not desirable, we could name this option "allow_jupyterlab_framework" as this is what it does. Maybe even more broadly it should be allowed_frameworks/blocked_frameworks list?

I like both. `allow_jupyterlab_framework` is the name I was actually looking for. 

>Maybe even more broadly it should be allowed_frameworks/blocked_frameworks list?

This would be nicer indeed with more flexibility. I prefer that.

Reply:
> Although documentation would be an excellent place for this, wdyt?

Agreed